src/controller/model/endpoint.ts

```diff
@@ -362,6 +362,26 @@ class Endpoint extends Entity {
     ): Promise<void> {
         const cluster = Zcl.Utils.getCluster(clusterKey);
         options = this.getOptionsWithDefaults(options, true, Zcl.Direction.CLIENT_TO_SERVER, cluster.manufacturerCode);
+
```

Comment:
Can we move this and the 2 similar blocks below to a function to prevent some code duplication?

Reply:
I tried but only read + write one are the same, the data we have in configureReporting is different.

Reply:
Actually after some cleanup I did I think I see a way to merge them. Let me try.

Reply:
Bah the map line is also different on all 3 -_-